modules/images/regenerate-existing-images/helper.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Functions used by the background process API.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Creates a background job.
+ *
+ * Job refers to the term and queue refers to the `background_job` taxonomy.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $name Name of job action.
+ * @param array  $data Data for the job.
+ * @return Perflab_Background_Job|WP_Error Job object if created successfully, else WP_Error.
+ */
+function perflab_create_background_job( $name, array $data = array() ) {
+	// Create the unique term name dynamically.
+	$term_name = 'job_' . time() . rand();
+	$term_data = wp_insert_term( $term_name, PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG );
+
+	if ( is_wp_error( $term_data ) ) {
+		return $term_data;
+	}
+
+	// Save the job data if present.
+	if ( ! empty( $data ) ) {
+		update_term_meta( $term_data['term_id'], Perflab_Background_Job::META_KEY_JOB_DATA, $data );
+	}
+
+	// Save job action. sanitize_title will be used before saving action.
+	update_term_meta( $term_data['term_id'], Perflab_Background_Job::META_KEY_JOB_ACTION, sanitize_title( $name ) );
```

Comment:
@felixarntz  I think its fine to mention in comment that action name can be alphanumeric only with underscores and dashes allowed because sanitize_title is being used. I've updated the comment.

Reply:
@ankitrox Wouldn't it be better to align on _one_ format? I don't have a strong preference on which one, but I think there should be a clear guidance on which one it should be.

Reply:
@felixarntz Let's proceed with underscore then because we have some other meta values with underscore too.